nixos/modules/services/hardware/powerd.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.powerd;
+in {
+  options = {
+    services.powerd = {
+      enable = mkEnableOption ''
+        powerd to handle battery power levels
+      '';
+      user = mkOption {
+        type = types.str;
+        default = "root";
+        description = ''
+          User under which powerd works.
+        '';
+      };
+      checkInterval = mkOption {
+        type = types.ints.unsigned;
+        default = 60;
+        description = ''
+          Sets the interval to check the battery level in seconds.
+        '';
+      };
+      warnOnLevel = mkOption {
+        type = types.ints.unsigned;
+        default = 10;
+        description = ''
+          Sets the battery level which will trigger a warn message.
+        '';
+      };
+      warnMessage = mkOption {
+        type = types.str;
+        default = "Battery level at 10%";
+        description = ''
+          Sets the warning message.
+        '';
+      };
+      execOnLevel = mkOption {
+        type = types.ints.unsigned;
+        default = 5;
+        description = ''
+          Sets the battery level which will trigger a command execution.
+        '';
+      };
+      execCommand = mkOption {
+        type = types.str;
+        default = "poweroff";
+        description = ''
+          Sets the exec command.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.powerd = {
+      description = ''
+        powerd - Simple daemon to handle battery power levels
+      '';
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.bash ];
+      script = ''
+          ${pkgs.powerd}/bin/powerd \
+            --check-interval ${toString cfg.checkInterval} \
+            --warn-on-level ${toString cfg.warnOnLevel} \
+            --warn-message "${cfg.warnMessage}" \
+            --exec-on-level ${toString cfg.execOnLevel} \
+            --exec-command "${cfg.execCommand}"
+      '';
+      serviceConfig = {
+        User = cfg.user;
```

Comment:
Hm, I don't like how this is root by default. How about this:
- The module sets `--exec-command` to be `echo TRIGGER > /run/powerd/triggerfile`
- In parallel to that a script that executes `execCommand` every time that file is written to
- This script can then run as root or whatever, while powerd can run with `DynamicUser = true` without any privileges. This way we know that only `execCommand` can be run as root
- This doesn't change the module interface at all, but is more secure

Reply:
@Infinisil Thanks! While testing I've found that dbus notifications don't work from systemd unit, so I'll remove the module. It will be better to run powerd right from user session.

Reply:
NixOS *does* support user systemd services with `systemd.user.services`. Want to use that instead? Can always be done later still if needed

Reply:
@Infinisil Yes I know, but are they popular? I've never used any because of this issue #21460